src/Nethermind/Nethermind.JsonRpc/Modules/Eth/SimulateTxExecutor.cs

```diff
@@ -120,91 +121,67 @@ public override ResultWrapper<IReadOnlyList<SimulateBlockResult>> Execute(
                 $"Too many blocks provided, node is configured to simulate up to {_blocksLimit} while {call.BlockStateCalls?.Count} were given",
                 ErrorCodes.InvalidParams);
 
-        secondsPerSlot ??= new BlocksConfig().SecondsPerSlot;
+        ulong secondsPerSlot = new BlocksConfig().SecondsPerSlot;
 
         if (call.BlockStateCalls is not null)
         {
-            long lastBlockNumber = -1;
-            ulong lastBlockTime = 0;
+            long lastBlockNumber = header.Number;
```

Comment:
Can we break down this setup to some meaningful helper methods? It is hard to follow the intent.

Reply:
Sure